Explorer/Assets/DCL/NameTags/NametagPlacementSystem.cs

```diff
@@ -54,28 +54,43 @@ protected override void Update(float t)
 
             UpdateTagQuery(World, camera);
             AddTagQuery(World, camera);
+            ProcessChatBubbleComponentsQuery(World);
         }
 
         [Query]
         [None(typeof(NametagView), typeof(PlayerComponent))]
         private void AddTag([Data] in CameraComponent camera, Entity e, in AvatarShapeComponent avatarShape, in CharacterTransform characterTransform, in PartitionComponent partitionComponent)
         {
-            if (partitionComponent.IsBehind) return;
+            if (partitionComponent.IsBehind || Vector3.Distance(camera.Camera.transform.position, characterTransform.Position) > nametagsData.maxDistance) return;
```

Comment:
Quick optimization, we could do
```C#
Vector3.DistanceSqr(camera.Camera.transform.position, characterTransform.Position) > nametagsData.maxDistance * nametagsData.maxDistance
```

Reply:
DistanceSqr doesn't exist 🤔

Reply:
moved to a function anyway

Reply:
haha sorry I mixed a few things, [distancesq](https://docs.unity3d.com/Packages/com.unity.mathematics@1.2/api/Unity.Mathematics.math.distancesq.html) is from the mathematics package.
In this case we would need to do `(B-A).distanceSqr > nametagsData.maxDistance * nametagsData.maxDistance`